packages/trimerge-sync/src/TrimergeClient.ts

```diff
@@ -228,12 +234,26 @@ export class TrimergeClient<
     return this.clientListSubs.subscribe(onChange, { origin: 'subscribe' });
   }
 
-  updateDoc(doc: LatestDoc, metadata: CommitMetadata, presence?: Presence) {
-    const ref = this.addNewCommit(doc, metadata, false);
-    this.setPresence(presence, ref);
-    this.mergeHeads();
-    this.docSubs.emitChange({ origin: 'self' });
-    this.sync();
+  updateDoc(
+    doc: LatestDoc,
+    metadata: CommitMetadata,
+    presence?: Presence,
+  ): Promise<void> {
+    return new Promise<void>((resolve, reject) => {
+      const ref = this.addNewCommit(
+        doc,
+        metadata,
+        false,
+        undefined,
+        undefined,
+        resolve,
+        reject,
+      );
+      this.setPresence(presence, ref);
+      this.mergeHeads();
+      this.docSubs.emitChange({ origin: 'self' });
+      this.sync();
+    });
```

Comment:
The control flow for resolving this promise is a little hard to read. `resolve` is handled by `addNewCommit` but we call a bunch of synchronous functions after `addNewCommit`. Does that promise get resolved without us calling those synchronous functions, or does it depend on them?

Reply:
Also, are we intentionally not awaiting the promise returned by `this.sync()`?

Reply:
Actually, that should clean up this whole thing if I just await the sync then we can avoid this whole stored resolve/reject nonsense.